File: 1067/main.cs
using System;
using System.Collections.Generic;

class Dir{
    public Dictionary<string, Dir> Subs = new Dictionary<string, Dir>();
}

class Program{
    
    static Dir[] dirs = new Dir[50001];
    static int p = 1;

    static Dir AddDir(Dir dir, string str)
    {
        if (!dir.Subs.ContainsKey(str))
            dir.Subs[str] = dirs[p++] = new Dir();
        return dir.Subs[str];
    }

    static void Print(Dir dir, int depth = 0)
    {
        foreach (var entry in dir.Subs)
        {
            Console.WriteLine(new string(' ', depth) + entry.Key);
            Print(entry.Value, depth + 1);
        }
    }

    static void Main()
    {
        for (int i = 0; i < dirs.Length; i++)
            dirs[i] = new Dir();

        int N = int.Parse(Console.ReadLine());
        for (int i = 0; i < N; i++)
        {
            string path = Console.ReadLine();
            string[] parts = path.Split('\\');
            Dir dir = dirs[0];
            foreach (var part in parts)
                dir = AddDir(dir, part);
        }

        Print(dirs[0]);
    }
}

File: 1067/main.py
class Dir:
    def __init__(self):
        self.subs = {}

dirs = [Dir() for _ in range(50001)]
p = 1

def add_dir(dir, name):
    global p
    if name not in dir.subs:
        dir.subs[name] = dirs[p]
        p += 1
    return dir.subs[name]

def print_dirs(dir, depth=0):
    for name in sorted(dir.subs):
        print(' ' * depth + name)
        print_dirs(dir.subs[name], depth + 1)

def main():
    N = int(input())
    for _ in range(N):
        path = input()
        parts = path.split('\\')
        dir = dirs[0]
        for part in parts:
            dir = add_dir(dir, part)

    print_dirs(dirs[0])

if __name__ == '__main__':
    main()
